### PR TITLE
mlm upgrade to new version format

### PR DESCRIPTION
**Pull request by najiyacl**
to_viz changes with mlm upgrade
This PR should go after the following
1.API Changes - https://bitbucket.oci.oraclecorp.com/projects/ODSC/repos/feature-store-dataplane/pull-requests/256/overview
2.UI Changes which would handle both new and old MLM response
3.MLM Upgrade

### CODE REVIEW

File: ads/feature_store/statistics/charts/frequency_distribution.py [hunk 1]
```diff
@@ -29,21 +30,22 @@ def __init__(self, frequency: List, bins: List):
...
+            bins=metric_data[0],
+            frequency=metric_data[1]
         )
 
     def add_to_figure(self, fig: Figure, xaxis: int, yaxis: int):
         xaxis_str, yaxis_str, x_str, y_str = self.get_x_y_str_axes(xaxis, yaxis)
         if (
-            type(self.frequency) == list
-            and type(self.bins) == list
-            and 0 < len(self.frequency) == len(self.bins) > 0
```
Comment (harsh97) [inline ads/feature_store/statistics/charts/frequency_distribution.py:46]: is length not equal anymore?

File: ads/feature_store/statistics/charts/probability_distribution.py [hunk 2]
```diff
@@ -29,21 +30,22 @@ def __init__(self, density: List, bins: List):
...
+            bins=metric_data[0],
+            density=metric_data[1]
         )
 
     def add_to_figure(self, fig: Figure, xaxis: int, yaxis: int):
         xaxis_str, yaxis_str, x_str, y_str = self.get_x_y_str_axes(xaxis, yaxis)
         if (
             type(self.density) == list
             and type(self.bins) == list
-            and 0 < len(self.density) == len(self.bins) > 0
```
Comment (harsh97) [inline ads/feature_store/statistics/charts/probability_distribution.py:46]: is length not equal anymore?

File: ads/feature_store/statistics/generic_feature_value.py [hunk 3]
```diff
@@ -19,6 +19,6 @@ def __validate__(self):
     def __from_json__(cls, json_dict: dict) -> "GenericFeatureValue":
         val = None
         if type(json_dict) == dict:
-            val = json_dict.get(cls.CONST_VALUE)
-
+            metric_data = json_dict.get(AbsFeatureValue.CONST_METRIC_DATA)
+            val = metric_data[0]
```
Comment (harsh97) [inline ads/feature_store/statistics/generic_feature_value.py:23]: should we check whether we got a list before accessing?

File: ads/feature_store/statistics/charts/probability_distribution.py [hunk 4]
```diff
@@ -29,21 +30,22 @@ def __init__(self, density: List, bins: List):
...
-        assert 0 < len(self.density) == len(self.bins) > 0
+        # assert 0 < len(self.density) == len(self.bins) > 0
 
     @classmethod
     def __from_json__(cls, json_dict: dict) -> "ProbabilityDistribution":
+        metric_data = json_dict.get(AbsFeatureValue.CONST_METRIC_DATA)
         return cls(
-            density=json_dict.get(ProbabilityDistribution.CONST_DENSITY),
-            bins=json_dict.get(ProbabilityDistribution.CONST_BINS),
+            bins=metric_data[0],
```
Comment (harsh97) [inline ads/feature_store/statistics/charts/probability_distribution.py:39]: should we check if we got a list before accessing?

**Comment by harsh97** [comment]
we should also handle the older MLM payload visualisation I feel.

File: ads/feature_store/statistics/charts/frequency_distribution.py (hunk 1 above)
Comment (najiyacl) [reply ads/feature_store/statistics/charts/frequency_distribution.py:46]: No

File: ads/feature_store/statistics/generic_feature_value.py (hunk 3 above)
Comment (najiyacl) [reply ads/feature_store/statistics/generic_feature_value.py:23]: Can be checked..but I see we have the exception handling

File: ads/feature_store/response/response_builder.py [hunk 5]
```diff
@@ -12,7 +12,7 @@ class ResponseBuilder(Builder):
 
     CONST_CONTENT = "content"
 
-    def __init__(self, content: str):
+    def __init__(self, content: str, version: int = 1):
```
Comment (KshitizLohia) [inline ads/feature_store/response/response_builder.py:15]: why the default version should be 1?

File: ads/feature_store/statistics/charts/box_plot.py [hunk 6]
```diff
@@ -44,13 +44,22 @@ def __init__(self, q1: float, q2: float, q3: float):
...
             return cls(
                 json_dict.get(cls.CONST_Q1),
                 json_dict.get(cls.CONST_Q2),
                 json_dict.get(cls.CONST_Q3),
             )
 
+        @classmethod
+        def __from_json_v2__(cls, json_dict: dict) -> "BoxPlot.Quartiles":
+            metric_data = json_dict.get(AbsFeatureValue.CONST_METRIC_DATA)
+            return cls(metric_data[0], metric_data[1], metric_data[2])
```
Comment (KshitizLohia) [inline ads/feature_store/statistics/charts/box_plot.py:61]: 1. is there no `q1`, `q2`.. in the newer schema? what happens if array does not have 0,1,2 as indices?
2. are we handling cases where json_dict does not contain `AbsFeatureValue.CONST_METRIC_DATA` outside of this method?

**Comment by KshitizLohia** [comment]
thanks @najiyacl for putting it up. this is nicely handled using version property. can we add screenshots of one execution with older and newer version to the PR?

File: ads/feature_store/statistics/charts/box_plot.py (hunk 6 above)
Comment (najiyacl) [reply ads/feature_store/statistics/charts/box_plot.py:61]: As per the existing code the exception is handled to ignore

File: ads/feature_store/response/response_builder.py (hunk 5 above)
Comment (najiyacl) [reply ads/feature_store/response/response_builder.py:15]: Currently this version is used only in stat and for version 2(new logic) the version is passed via Statistics object

**Review by KshitizLohia — APPROVED** [review]

**Pull request merged**